src/functions/indAffine.jl

```diff
@@ -26,11 +26,11 @@ immutable IndAffine{T <: RealOrComplex} <: IndicatorConvex
   end
 end
 
-IndAffine{T <: RealOrComplex}(A::AbstractArray{T,2}, b::AbstractArray{T,1}) =
-  IndAffine{T}(A, b)
+IndAffine{T <: RealOrComplex, M<:AbstractArray{T,2}, V<:AbstractArray{T,1}}(A::M, b::V) =
+  IndAffine{T,M,V}(A, b)
```

Comment:
I used some of these outer constructors to avoid the following situation:
```julia
julia> immutable MyType{R <: Real}
           a::R
           function MyType(a::R)
               if a <= 0 error("a must be positive") end
           end
       end

julia> b = MyType(Float64(5.0))
ERROR: MethodError: Cannot `convert` an object of type Float64 to an object of type MyType{R<:Real}
This may have arisen from a call to the constructor MyType{R<:Real}(...),
since type constructors fall back to convert methods.
 in MyType{R<:Real}(::Float64) at ./sysimg.jl:53

julia> b = MyType{Float64}(Float64(5.0)) # this one is OK but ugly to use

julia> MyType{R <: Real}(a::R) = MyType{R}(a) # define outer constructor
MyType{R<:Real}

julia> b = MyType(Float64(5.0)) # now this is also OK
``` 
Is there a better way to do this? Has something changed wrt this in 0.6?